discv4/kademlia.go

```diff
@@ -0,0 +1,133 @@
+package discv4
+
+import (
+	"container/list"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/indexsupply/lib/enr"
+)
+
+const (
+	maxBucketSize  = 16
+	addrByteSize   = 256 // size in bytes of the node ID
+	bucketsCount   = 20  // very rare we will ever encounter a node closer than log distance 20 away
+	minLogDistance = addrByteSize + 1 - bucketsCount
+)
+
+type kademliaTable struct {
```

Comment:
what do you think of making a kademlia package and then rename this type to `Table`